mesheryctl/internal/cli/root/system/model.go

```diff
@@ -0,0 +1,364 @@
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return err
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 0 {
+			return errors.New(utils.SystemModelSubError("this command takes no arguments\n", "list"))
+		}
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			log.Fatalln(err, "error processing config")
+		}
+
+		baseUrl := mctlCfg.GetBaseMesheryURL()
+		var url string
+		if cmd.Flags().Changed("page") {
+			url = fmt.Sprintf("%s/api/meshmodels/models?page=%d", baseUrl, pageNumberFlag)
+		} else {
+			url = fmt.Sprintf("%s/api/meshmodels/models?pagesize=all", baseUrl)
+		}
+		req, err := utils.NewRequest(http.MethodGet, url, nil)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		resp, err := utils.MakeRequest(req)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		// defers the closing of the response body after its use, ensuring that the resources are properly released.
+		defer resp.Body.Close()
+
+		data, err := io.ReadAll(resp.Body)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		modelsResponse := &models.MeshmodelsAPIResponse{}
+		err = json.Unmarshal(data, modelsResponse)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		header := []string{"Category", "Model", "Version"}
+		rows := [][]string{}
+
+		for _, model := range modelsResponse.Models {
+			if len(model.DisplayName) > 0 {
+				rows = append(rows, []string{model.Category.Name, model.Name, model.Version})
+			}
+		}
+
+		if len(rows) == 0 {
+			// if no model is found
+			utils.Log.Info("No model(s) found")
+		} else {
+			utils.PrintToTable(header, rows)
+		}
+
+		return nil
+	},
+}
+
+// represents the `mesheryctl system model view [model-name]` subcommand.
+var viewModelCmd = &cobra.Command{
+	Use:   "view",
+	Short: "view model",
+	Long:  "view a model queried by its name",
+	Example: `
+// View current provider
+mesheryctl system model view [model-name]
+	`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		//Check prerequisite
+		hcOptions := &HealthCheckOptions{
+			IsPreRunE:  true,
+			PrintLogs:  false,
+			Subcommand: cmd.Use,
+		}
+		hc, err := NewHealthChecker(hcOptions)
+		if err != nil {
+			return ErrHealthCheckFailed(err)
+		}
+		// execute healthchecks
+		err = hc.RunPreflightHealthChecks()
+		if err != nil {
+			cmd.SilenceUsage = true
+			return err
+		}
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return err
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+	Args: func(_ *cobra.Command, args []string) error {
+		const errMsg = "Usage: mesheryctl system model view [model-name]\nRun 'mesheryctl system model view [model-name]' to view the model"
```

Comment:
One line of usage for two subcommands isn't enough documentation or example, @zakisk

Reply:
Still not answered...